pipeline/test_etl.py

```diff
@@ -0,0 +1,232 @@
+                "genres": ["Rock", "Pop"],
+                "release_date": "01-10-2021",
+                "sale_date": "2021-10-01",
+            }
+        ]
+        df = create_sales_dataframe(mock_sales_info)
+        assert isinstance(df, pd.DataFrame)
+        assert len(df) == 1
+        assert "release_type" in df.columns
+        assert df.iloc[0]["release_type"] == "t"
+
+
+class TestLoad:
+    """Tests for the load phase of the ETL process."""
+
+    @pytest.fixture(autouse=True)
+    def mock_env(self, monkeypatch):
+        """Mock database environment variables."""
+        monkeypatch.setenv("DB_HOST", "mock_host")
+        monkeypatch.setenv("DB_PORT", "5432")
+        monkeypatch.setenv("DB_USER", "mock_user")
+        monkeypatch.setenv("DB_PASSWORD", "mock_password")
+        monkeypatch.setenv("DB_NAME", "mock_db")
+
+    @pytest.fixture
+    def mock_connection(self):
+        """Provide a mocked database connection."""
+        return MagicMock()
+
+    @pytest.fixture
+    def mock_cursor(self, mock_connection):
+        """Provide a mocked database cursor."""
+        cursor = MagicMock()
+        mock_connection.cursor.return_value = cursor
+        return cursor
+
+    def test_load_get_connection(self):
+        """Ensure database connection is established."""
+        with patch("psycopg2.connect") as mock_connect:
+            connection = get_connection()
+            assert mock_connect.called
+            assert connection is not None
+
+    def test_load_get_cursor(self, mock_connection):
+        """Ensure cursor is retrieved from the connection."""
+        cursor = get_cursor(mock_connection)
+        assert cursor is not None
+        mock_connection.cursor.assert_called_once()
+
+
+    def test_load_insert_country_existing(self, mock_cursor):
+        """Test insertion of an existing country."""
+        mock_cursor.fetchone.return_value = (True,)
+        insert_country("TestCountry", mock_cursor)
+        mock_cursor.execute.assert_called_once_with(
+            "SELECT EXISTS (SELECT 1 FROM country WHERE country_name = %s);", ("TestCountry",)
+        )
+
+    def test_load_insert_country_new(self, mock_cursor):
+        """Test insertion of a new country."""
+        mock_cursor.fetchone.return_value = (False,)
+        insert_country("NewCountry", mock_cursor)
+        mock_cursor.execute.assert_any_call(
+            "INSERT INTO country (country_name) VALUES (%s);", ("NewCountry",)
+        )
+
+    def test_load_insert_artist_new(self, mock_cursor):
+        """Test insertion of a new artist."""
+        mock_cursor.fetchone.return_value = None
+        insert_artist("NewArtist", mock_cursor)
+        assert mock_cursor.execute.call_count == 2
+
+    def test_load_insert_genres_new(self, mock_cursor):
+        """Test insertion of a new genre."""
+        mock_cursor.fetchone.side_effect = [None, (1,)]
+        genre_id = insert_genres("NewGenre", mock_cursor)
+        assert genre_id == 1
```

Comment:
Could we add a test for get_id_from_table() to ensure correct ID's are retrieved or returns None